### DISCUSSION TITLE
[Bug] Success Rate is Incorrect with Vert.x

### DISCUSSION THREAD

**Discussion by LangInteger** [apache/skywalking#11674] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

The success rate is incorrect when Skywalking is working with a Vert.x web server.

This [demo project](https://github.com/LangInteger/vertx-skywalking) contains two APIs:

- 127.0.0.1:8081/success responds with HTTP code 200
- 127.0.0.1:8081/fail responds with HTTP code 500

In the dashboard, the success rate for both is 100%.

![](https://raw.githubusercontent.com/LangInteger/vertx-skywalking/main/pictures/dashboard.jpeg)

### 2.1 Start ES/OAP/UI/Vert.x_Web_Server

Using the following command:

```shell
wget -nc https://dlcdn.apache.org/skywalking/java-agent/9.1.0/apache-skywalking-java-agent-9.1.0.tgz &&
tar xzvf ./apache-skywalking-java-agent-9.1.0.tgz &&
./gradlew clean build && \
docker-compose build && \
export OAP_IMAGE=apache/skywalking-oap-server:9.7.0 && \
export UI_IMAGE=apache/skywalking-ui:9.7.0 && \
export ES_VERSION=7.10.2 && \
docker-compose up
```

### What you expected to happen

The success rate for the `fail` API should be 0 as it responds with HTTP code 500.

### How to reproduce

- Download the demo project: https://github.com/LangInteger/vertx-skywalking
- Follow the `Reproduce` section in the README.md file

### Anything else

Please kindly let me know if it is not a bug and I can correct it with some settings. Thanks!

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Success rate and traffic is calculated based on entry spans from your traces. Please check there.

**Reply by LangInteger** [reply]
Do you mean it does not treat the HTTP response with code other than 200 as a failed API call by default?
Could you please share a `get-started` doc to set up this feature (I want the success rate to reflect the HTTP code meaning)?

**Reply by LangInteger** [reply]
I set up the `entry span` as described here:

- https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/application-toolkit-tracer/

![image](https://github.com/apache/skywalking/assets/17768055/7f065e9b-f79b-46b4-b495-a0df78a68311)

The result now is a combination of the built-in endpoint and the endpoint representing the span created by me manually, which is better but still, I want either:

- I can directly change the success rate of the build-in endpoint

or

- set up span manually in code and override the corresponding build-in endpoint

**Reply by wu-sheng** [reply]
I didn't ask you to create a new span, my point is this rate is determined by the span created by existing plugin. 
If it is counted twice, it means, you have two plugins for one request, you should remove one.

**Reply by LangInteger** [reply]
The `POST:/fail` and `POST:/success` are not controlled by my code.

Simply start the jar file with the agent `-javaagent:skywalking-agent/skywalking-agent.jar` and then they are reported to the OAP server. How can I control this behavior?

**Reply by LangInteger** [reply]
Or do you mean this needs some work from the `Vert.x plugin` side as it seems like this is the default behavior of that plugin.

**Reply by wu-sheng** [reply]
I don't know plugins, they are too many. My point is, yes, that is controlled by it.

**Reply by LangInteger** [reply]
Thanks. I will try to contact the original plugin author.

**Reply by LangInteger** [reply]
Hi Sheng, it seems like Brandon is not actively responding currently.
I would appreciate your letting me know where the plugin-related code is stored. 
I cannot find these code now:
- https://github.com/apache/skywalking/pull/2386

**Reply by wu-sheng** [reply]
It is Christmas vacations, he should not be on Github.

The codes in the same place, but in this new repo, https://github.com/apache/skywalking-java

**Reply by LangInteger** [reply]
@wu-sheng Thank you! I will look at this when I have time.

**Reply by BFergerson** [reply, marked as answer]
Apologies for the wait. Please check: https://github.com/apache/skywalking-java/pull/662

**Comment by LangInteger** [comment]
@BFergerson Hi, Brandon, do you have any insights on this case?
